### PR TITLE
Remove abi flag

### PR DESCRIPTION
**Pull request by qicosmos**
## Why are these changes needed?

Remove the ABI flag: _GLIBCXX_USE_CXX11_ABI.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by mwtian** [comment]
Still looking into why this fails.

File: ci/travis/ci.sh [hunk 1]
```diff
@@ -206,8 +206,11 @@ test_cpp() {
   bazel test //cpp:cluster_mode_test --test_arg=--external_cluster=true --test_arg=--redis_password="1234" \
     --test_arg=--ray_redis_password="1234"
 
-  # run the cpp example
-  cd cpp/example && sh run.sh
+  # avoid abi issue.
```
Comment (raulchen) [inline ci/travis/ci.sh:209]: Add a TODO, and explain why we need to skip this test in CI.

File: ci/travis/ci.sh [hunk 2]
```diff
@@ -199,6 +199,7 @@ test_python() {
 }
 
 test_cpp() {
+  echo build --cxxopt="-D_GLIBCXX_USE_CXX11_ABI=0" >> ~/.bazelrc
```
Comment (raulchen) [inline ci/travis/ci.sh:202]: Explain why need this

**Review by raulchen — APPROVED** [review]

**Comment by mwtian** [comment]
Thanks for this change!

**Pull request merged**